pandas/tests/io/parser/test_dtypes.py

```diff
@@ -550,3 +550,39 @@ def test_numeric_dtype(all_parsers, dtype):
 
     result = parser.read_csv(StringIO(data), header=None, dtype=dtype)
     tm.assert_frame_equal(expected, result)
+
+
+def test_boolean_dtype(all_parsers):
+    parser = all_parsers
+    data = "\n".join(
+        [
+            "a",
+            "True",
+            "TRUE",
+            "true",
+            "False",
+            "FALSE",
+            "false",
+            "NaN",
+            "nan",
+            "NA",
+            "null",
+            "NULL",
+        ]
+    )
+
+    assert all(s in data for s in ["True", "TRUE", "true"])
+    assert all(s in data for s in ["False", "FALSE", "false"])
+    assert all(s in data for s in ["NaN", "nan", "NA", "null", "NULL"])
```

Comment:
Are those asserts needed? (I think it is clear they are in the data, as you generated it yourself?) 
I think the comparison with result of read_csv below is good enough

Reply:
Agreed.